lib/browser/api/navigation-controller.js

```diff
@@ -62,6 +62,25 @@ var NavigationController = (function () {
       options = {}
     }
     this.pendingIndex = -1
+    if (options.hasOwnProperty('extraHeaders')) {
+      let extraHeaders = options['extraHeaders']
+      if (typeof extraHeaders === 'object') {
+        let newExtraHeaders = []
+        for (let key in extraHeaders) {
+          if (extraHeaders.hasOwnProperty(key)) {
+            newExtraHeaders.push([key, extraHeaders[key]])
+          }
+        }
+        extraHeaders = newExtraHeaders
+      }
+      if (Array.isArray(extraHeaders)) {
+        // Escape header values for security:
+        //   replace LWS with a single SP
+        //   remove CR and LF since the char is invalid here
+        extraHeaders = extraHeaders.map(pair => pair.map(v => v.replace(/\015\012[\040\011]/g, ' ').replace(/[\015\012]/g, '')).join(': ')).join('\n')
```

Comment:
This is a lot for single line, would you mind reformatting this to break it up across multiple lines to make it easier to see how the object is being transformed.